src/ert/run_models/ensemble_experiment.py

```diff
@@ -61,7 +61,35 @@ def run_experiment(
         restart: bool = False,
     ) -> None:
         self.log_at_startup()
-        if not restart:
+        # If design matrix is present, we substitute the experiment parameters
+        # with those in the design matrix
+        if self.ert_config.analysis_config.design_matrix is not None:
+            if (
+                self.ert_config.analysis_config.design_matrix.parameter_configuration
+                is None
+            ):
+                self.ert_config.analysis_config.design_matrix.read_design_matrix()
+            assert (
```

Comment:
Could we get rid of this assert and make `read_design_matrix` responsible for ensuring that `design_matrix.parameter_configuration` is not None?

Reply:
mypy was complaining ...